File: 프로그래머스/Level.1/17. 콜라츠 추측.js
function solution(num) {
  for (let i = 0; i < 500; i++) {
    if (num != 1) {
      num = num % 2 === 0 ? num / 2 : num * 3 + 1;
    } else {
      return i;
    }
  }
  return -1;
}

solution(6);

/////다른 풀이

function collatz(num) {
  var answer = 0;

  while (num != 1 && answer != 500) {
    num % 2 == 0 ? (num = num / 2) : (num = num * 3 + 1);
    answer++;
  }

  return num == 1 ? answer : -1;
}

File: 프로그래머스/Level.1/17. 콜라츠 추측.py
# 나의 풀이
def solution(num):
    count = 0

    while num > 1 and count < 500:
        if num % 2 == 0:
            num = num // 2
        elif num % 2 == 1:
            num = num * 3 + 1
        count += 1

    return (count if num == 1 else -1)

# 팀 코드
def solution(num):
    for i in range(500):
        if num == 1:
            return i
        if (num%2==0):
            num /=2 
        else:
            num = num*3+1
    return -1